Initial implementation of MessageBox.

raghucssit opened the pull request.

tmssngr commented:
Thanks, looks good. I've added two fixup commits and rebased. If you agree, please feel free to squash them and merge.

raghucssit commented:
> Thanks, looks good. I've added two fixup commits and rebased. If you agree, please feel free to squash them and merge.

Thank you for the fixes.. All changes looks good...
